library/azure_rm_webapp_slot.py

```diff
@@ -0,0 +1,993 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_webapp_slot
+version_added: "2.8"
+short_description: Manage Web App slot.
+description:
+    - Create, update and delete Web App slot.
+
+options:
+    resource_group:
+        description:
+            - Name of the resource group to which the resource belongs.
+        required: True
+    name:
+        description:
+            - Unique name of the deployment slot to create or update.
+        required: True
+    webapp_name:
+        description:
+            - Web app name which this deployment slot belongs to.
+        required: True
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+    configuration_source:
+        description:
+            - Source slot to clone configurations from. Use webapp's name to refer to the production slot.
+    auto_swap_slot_name:
+        description:
+            - Target slot name to auto swap.
+            - Set it to 'None' to disable auto slot swap.
```

Comment:
What does it mean set it to "None"? should be string?

I think we should avoid using "none"